webapp/channels/src/components/threading/thread_viewer/thread_viewer.tsx

```diff
@@ -49,6 +49,7 @@ export type Props = Attrs & {
     selectedPostFocusedAt?: number;
     isThreadView?: boolean;
     inputPlaceholder?: string;
+    rootPostId: string;
```

Comment:
Looking at the implementation, it seems that `selected.id` and `rootPostId` should always be the same, right? If that is the case, why not use always `rootPostId`(now that we are passing it into this component)? That will simplify a lot of the changes.

There may be some places like the `componentDidUpdate` checks where you still want to compare over selected (to take into consideration when it changes from undefined to actually being there), but the places where you only want the id, I think we can use directly `rootPostId` instead of `this.props.selected?.id || this.props.rootPostId`.

Reply:
It should be but I wrote it like this to be safe.

Reply:
Ideally, I feel like we wouldn't check both of those for IDs since it introduces a potential risk of those getting desynced somehow, but since that that should never happen (unless something's REALLY gone wrong), this seems fine to me. While checking both feels odd, it does seem like the best way to ensure the current code works correctly